examples/platforms/simulation/radio.c

```diff
@@ -164,6 +164,8 @@ static otMacKeyMaterial sCurrKey;
 static otMacKeyMaterial sNextKey;
 static otRadioKeyType   sKeyType;
 
+static bool sDeniedNodeIds[MAX_NETWORK_SIZE];
```

Comment:
```suggestion
static uint8_t sDeniedNodeIds[(MAX_NETWORK_SIZE + 7) / 8];
```
nit: use less ram

Reply:
I did not use a bit to store a single state but a byte. `std::bitset` / `ot::BitVector` could not be utilized here as it is not a C++ source file. If a bit array were to be used, I would have to implement something similar to `ot::BitVector` in C.

Thoughts on whether to switch from byte array to bit array?

Reply:
I still think bit-array is better. 
When the network size is 999, the memory consumption is 999B v.s. 125B. 

> If a bit array were to be used, I would have to implement something similar to ot::BitVector in C.

I think we should transform simulation platform code from C to C++. But that's for another PR.

Reply:
I switched it from a byte array to a bit array. Then I implemented two functions `NodeIdFilterIsConnectable` and `NodeIdFilterDeny` as getter and setter in C.